iot/api-client/manager/src/main/java/com/example/cloud/iot/examples/DeviceRegistryExample.java

```diff
@@ -173,6 +173,103 @@ public static void deleteRegistry(String cloudRegion, String projectId, String r
   }
   // [END iot_delete_registry]
 
+  // [START iot_clear_registry]
+  /** Delete this registry from Cloud IoT. */
+  public static void clearRegistry(String cloudRegion, String projectId, String registryName)
+      throws GeneralSecurityException, IOException {
+    GoogleCredential credential =
+        GoogleCredential.getApplicationDefault().createScoped(CloudIotScopes.all());
+    JsonFactory jsonFactory = JacksonFactory.getDefaultInstance();
+    HttpRequestInitializer init = new RetryHttpInitializerWrapper(credential);
+    final CloudIot service =
+        new CloudIot.Builder(GoogleNetHttpTransport.newTrustedTransport(), jsonFactory, init)
+            .setApplicationName(APP_NAME)
+            .build();
+    final String registryPath =
+        String.format(
+            "projects/%s/locations/%s/registries/%s", projectId, cloudRegion, registryName);
+
```

Comment:
Pseudo code for what's going on?
```
ForAll gateways
  ForAll deviceNumIds
     UnbindDeviceFromGatewayRequest

ForAll Devices
  Delete DevicePath

DeleteRegistry
```

Reply:
Yes, this is high-level what's going on.